### ISSUE TITLE
setting `machine.files` to create empty `/data/traefik/.keep` file causes boot-loop

### ISSUE THREAD

**Issue by jokeyrhyme** [siderolabs/talos#5147]
## Bug Report

I used `talosctl edit machineconfig` to change  `machine.files` so that talos would create an empty `/data/traefik/.keep` file

Once the configuration applied, the node entered a boot loop and does not respond to `talosctl` commands, and only responds to `ping` for maybe 20-30 seconds before rebooting again

Earlier today, I changed `machine.files` in a similar way, but I did it to all the nodes in my cluster, so unfortunately the entire cluster became unresponsive

At first I believed this was due to installing the CSIDriver for CephFS, but I've wiped my cluster nodes and started bringing up my desired setup, and it is changing `machine.files` that seems to be associated with a node becoming non-responsive, as this time I have not even touched cephfs-csi-cephfs yet

### Description

My configuration is the default in terms of disks and filesystems, there are no additional disks or mounted filesystems

```
machine:
  files:
    - content: '' # The contents of the file.
      permissions: 0o666 # The file's permissions in octal.
      path: /data/traefik/.keep # The path of the file.
      op: create # The operation to use
```

### Logs

Unable to retrieve logs, node is non-responsive

### Environment

- Talos version: captured from other nodes in my cluster (I only applied the new `machine.files` to a single node this time)

```
Client:
        Tag:         v0.14.3
        SHA:         73f4efd7b53baac12709281552f4eed7b47bfc1e
        Built:
        Go version:  go1.17.7
        OS/Arch:     linux/amd64
Server:
        NODE:        192.168.1.215
        Tag:         v1.0.0-beta.1
        SHA:         602e049d
        Built:
        Go version:  go1.17.8
        OS/Arch:     linux/arm64
        Enabled:     RBAC
```

- Kubernetes version:

```
Client Version: v1.23.4
Server Version: v1.23.1
```

- Platform:

```
Linux hellmouth 5.16.15-arch1-1 #1 SMP PREEMPT Thu, 17 Mar 2022 00:30:09 +0000 x86_64 GNU/Linux
```

**Comment by smira** [comment]
we have a ticket #4669 in progress which should help with this issue.

Talos root file system is read-only, so if you want to create some directory, it should go under `/var` (EPHERMAL partition) which is `rw`

**Comment by jokeyrhyme** [comment]
Ah, I thought it would be something like this, thanks!

Is it worth updating the "files" section under https://www.talos.dev/docs/v0.14/reference/configuration/#machineconfig to warn about this?

Is this something that could be detected/validated during `talosctl edit machineconfig` so that invalid configurations like this are blocked?

**Comment by smira** [comment]
we should definitely improve the docs, it's hard to validate the actual files completely, as it might depend on mounted filesystems.

the fix in #4669 allows to fix the machine config if it can't be applied
